design/mon-health.md

```diff
@@ -0,0 +1,66 @@
+The identity of a mon is its IP address.
+
+To have an immutable IP address in Kubernetes, Rook creates a K8s service for each monitor. The clusterIP of the service will act as the stable identity. 
+
+When a monitor pod starts, it will bind to its podIP and it will expect communication to be via its service IP address.
+
+## Monitor Quorum
+Multiple mons work together to provide redundancy by each keeping a copy of the metadata. A variation of the distributed algorithm Paxos is used to establish 
+consensus about the state of the cluster. Paxos requires a super-majority of mons to be running in order to establish quorum and perform operations in the cluster. 
+If the majority of mons are not running, quorum is lost and nothing can be done in the cluster. 
+
+### How many mons?
+Most commonly a cluster will have three mons. This would mean that one mon could go down and allow the cluster to remain healthy. 
+You would still have 2/3 mons running to give you consensus in the cluster for any operation.
+
+You will always want an odd number of mons. Fifty percent of mons will not be sufficient to maintain quorum. Ff you had two mons and one of them went down, 
+you would have 1/2 of quorum. Since that is not a super-majority, the cluster would have to wait until the second mon is up again.
+
+The number of mons to create in a cluster depends on your tolerance for losing a node. If you have 1 mon zero nodes can be lost to maintain quorum. 
+With 3 mons one node can be lost, and with 5 mons two nodes can be lost. Because the Rook operator will automatically start a new new monitor if one dies, 
+you typically only need three mons. The more mons you have, the more overhead there will be to make a change to the cluster, which could become a performance issue in a large cluster. 
+
+## Mitigating Monitor Failure
+Whatever the reason that a mon may fail (power failure, software crash, software hang, etc), there are several layers of mitigation in place to help recover the mon. It is
+always better to bring an existing mon back up than to failover to bring up a new mon. 
+
+The Rook operator creates a mon with a ReplicaSet to ensure that the mon pod will always be restarted if it fails. If a mon pod stops for any reason, Kubernetes will 
+automatically start the pod up again.
+
+In order for a mon to support a pod/node restart, you will need to set the `dataDirHostPath` so the mon metadata will be persisted to disk in this folder. 
+This will allow the mon to start back up with its existing metadata and continue where it left off even if the pod had to be re-created. Without this host path, the mon cannot
+start again.
+
+## Failing over a Monitor
+If the mitigating steps for mon failure don't bring a mon back up after some period of time (five minutes), the operator will make the decision to terminate the old monitor pod
```

Comment:
I think it is worth mentioning that the mon back up time is configurable by a flag.